core/src/main/java/org/elasticsearch/action/support/replication/ReplicationRequest.java

```diff
@@ -177,6 +177,11 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeString(index);
     }
 
+    @Override
+    public Task createTask(long id, String type, String action) {
+        return new ReplicationTask(id, type, action, this::getDescription, getParentTaskNode(), getParentTaskId());
```

Comment:
I didn't realize we are always creating a task (since the task management infra supports null tasks, I gathered this is opt in). I had a more careful look and the description is indeed evaluated on demand. Benchmarks also look OK. Thanks for explaining.